src/sonic_ax_impl/mibs/vendor/cisco/ciscoEntityFRUControlMIB.py

```diff
@@ -0,0 +1,106 @@
+import subprocess
+import re
+
+from sonic_ax_impl import mibs
+from ax_interface import MIBMeta, ValueType, MIBUpdater, MIBEntry, SubtreeMIBEntry
+from ax_interface.encodings import ObjectIdentifier
+
+import pdb
+
+class PowerStatusHandler:
+    """
+    Class to handle the SNMP request
+    """
+    def __init__(self):
+        """
+        init the handler
+        """
+
+        # Holds the number of supported PSU
+        self.num_psus = self._getNumOfSupportedPsu()
+
+    def _getNumOfSupportedPsu(self):
+        """
+        :return: the number of supported PSU
+        """
+        try:
+            output = subprocess.check_output(['psuutil', 'numpsus'])
+            if str(output[0]).isdigit():
+                return int(output)
+        except subprocess.CalledProcessError:
+            mibs.logger.error("failed to run psuutil tool")
+        except IndexError:
+            mibs.logger.error("failed to get he number of supported PSU")
+
+        return 0
+
+    def _getPsuIndex(self, sub_id):
+        """
+        Get the PSU index from sub_id
+        :return: the index of supported PSU
+        """
+        if not sub_id or len(sub_id) > 1:
+            return None
+
+        psu_index = int(sub_id[0])
+
+        if psu_index < 1 or psu_index > self.num_psus:
+            return None
+
+        return psu_index
+
+    def get_next(self, sub_id):
+        """
+        :param sub_id: The 1-based snmp sub-identifier query.
+        :return: the next sub id.
+        """
+        if not sub_id:
+            return (1,)
+
+        psu_index = self._getPsuIndex(sub_id)
+
+        if psu_index and psu_index + 1 <= self.num_psus:
+            return (psu_index + 1,)
+
+        return None
+
+    def getPsuStatus(self, sub_id):
+        """
+        :param sub_id: The 1-based sub-identifier query.
+        :return: the status of requested PSU.
+        """
+        psu_index = self._getPsuIndex(sub_id)
+
+        if not psu_index:
+            return None
+
+        output = ""
+        try:
+            output = subprocess.check_output(['psuutil', 'status', '-i', str(psu_index), '--textonly'])
+        except subprocess.CalledProcessError:
+            mibs.logger.error("failed to get PSU status from psuutil tool")
+            return None
+
+        psu_status = 0
+        status_list = str(output).split(':')
```

Comment:
Thanks. Please add the sample as code comment.
Also suggest checking len(status_list) instead of exception.

---
In reply to: [153851846](https://github.com/andrii-savka/sonic-snmpagent/pull/2#discussion_r153851846) [](ancestors = 153851846)